src/components/map/Layers/Countries.tsx

```diff
@@ -1,133 +1,156 @@
                     localEstimate: country.localEstimate,
                     nationalEstimate: country.nationalEstimate,
                     regionalEstimate: country.regionalEstimate,
                     sublocalEstimate: country.sublocalEstimate,
+                    partnershipConfig: partnershipConfig
                 }
             );
         }
     });
 }
 
-// Filters out Country features based on their existance within the new estimate grade prevalences
-function FilterCountryEstimates(map: mapboxgl.Map, estimateGradePrevalences: EstimateGradePrevalence[]) {
-    const onlyIso3: (string | null)[] = estimateGradePrevalences.map((c) => { return c.alpha3Code })
-
-    if (map.getLayer("Countries") && map.isSourceLoaded('Countries')) {
-        map?.setFilter('Countries',
-            ["in",
-                ["get", 'CODE'],
-                ["literal", onlyIso3]
-            ]);
-    }
-    else { // If countries has not been loaded, wait until it's source has finished.
-        map.on('sourcedata', function (e: any) {
-            if (e.sourceId === 'Countries' && map.isSourceLoaded('Countries')) {
-                map?.setFilter('Countries',
-                    ["in",
-                        ["get", 'CODE'],
-                        ["literal", onlyIso3]
-                    ]);
-            }
+function SetStudiesMode(map: mapboxgl.Map, iso3Name: string) {
+    map.removeFeatureState({
+        source: COUNTRY_LAYER_ID,
+        sourceLayer: COUNTRY_LAYER_ID
         });
-    }
+    map.setFeatureState(
+        {
+            source: COUNTRY_LAYER_ID,
+            sourceLayer: COUNTRY_LAYER_ID,
+            id: iso3Name,
+        },
+        {
+            isHighlighted: true
+        }
+    );
 }
 
-const Countries = (map: mapboxgl.Map | undefined, estimateGradePrevalences: EstimateGradePrevalence[]) => {
+const Countries = (map: mapboxgl.Map | undefined, {estimateGradePrevalences, countryFocus}: CountriesMapConfig) => {
 
     const [state] = useContext(AppContext);
     const [popup, setPopup] = useState<mapboxgl.Popup | undefined>(undefined);
-    const [currentFeatureId, setCurrentFeatureId] = useState<string | undefined>(undefined);
+    const [highlight, setHighlight] = useState<string | undefined>(undefined);
+    const history = useHistory();
 
     // If estimates are updated, waits until map is loaded then maps estimate data to country features
     useEffect(() => {
-        if (estimateGradePrevalences.length > 0 && map && map.getSource('Countries')) {
-            SetCountryEstimates(map, estimateGradePrevalences);
-        }
-        else if (map) {
-            map.on('sourcedata', function (e: any) {
-                if (e.sourceId === 'Countries' && map.isSourceLoaded('Countries')) {
+
+        if (map)
+        {
+            if(estimateGradePrevalences)
+            {
+                const sourceLoaded = map.getSource(COUNTRY_LAYER_ID) !== undefined
+                if (!sourceLoaded) 
```

Comment:
are we okay with `map.getSource(COUNTRY_LAYER_ID) == null` in this case? if not, i think we can just do a null | undefined check this way 
```suggestion
                if (!map.getSource(COUNTRY_LAYER_ID) 
```